nengo_dl/compat.py

```diff
@@ -14,6 +14,24 @@
 from packaging import version
 from tensorflow.python.eager import context
 
+
+class NoType:
+    """A type that can never be instantiated."""
+
+    def __init__(self, *arg, **kwargs):  # pragma: no cover
+        raise RuntimeError("Cannot instantiate")
+
+
+def make_dummy_type(name):
+    """Return a NoType subclass with the given name."""
+    return type(name, (NoType,), {})
+
+
+def is_dummy_type(t):
```

Comment:
I think we can remove this function after my other suggestions (re: checking nengo version rather than the type).